### PR TITLE
PHOENIX-5185 support Math PI function

### PR DESCRIPTION
**Pull request by yanxinyi**

### CODE REVIEW

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/MathPIFunctionEnd2EndIT.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+        }
+        if (Math.abs(a - b) <= ZERO) {
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    @Test
+    public void testMathPIFunction() throws Exception {
```
Comment (karanmehta93) [inline phoenix-core/src/it/java/org/apache/phoenix/end2end/MathPIFunctionEnd2EndIT.java]: nit: please break this up in 3 tests

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/MathPIFunctionEnd2EndIT.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+import org.apache.phoenix.expression.function.MathPIFunction;
+import org.junit.Test;
+
+/**
+ * End to end tests for {@link MathPIFunction}
+ */
+public class MathPIFunctionEnd2EndIT extends ParallelStatsDisabledIT {
+    private static final double ZERO = 1e-9;
+
+    private static boolean twoDoubleEquals(double a, double b) {
```
Comment (karanmehta93) [inline phoenix-core/src/it/java/org/apache/phoenix/end2end/MathPIFunctionEnd2EndIT.java]: nit: You might wanna move it in a utility class and use it for other tests too.

**Comment by karanmehta93** [comment]
Looks pretty good overall, Nice work @yanxinyi !

**Comment by yanxinyi** [comment]
1. split into 3 test cases.
2. moved utility function to BaseTest
3. removed dup utility function on all math function test files and called from utility function inside BaseTest

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/MathPIFunctionEnd2EndIT.java (hunk 1 above)
Comment (karanmehta93) [reply phoenix-core/src/it/java/org/apache/phoenix/end2end/MathPIFunctionEnd2EndIT.java]: Sorry, I wasn't much clear earlier. Please put this in three separate Junit `@Test` tests.

**Comment by karanmehta93** [comment]
Lets get your PI function patch committed on the PI day. :-)

**Comment by karanmehta93** [comment]
Thanks @yanxinyi for the patch. Can you also put up PR for 4.x-HBase-1.4, 4.x-HBase-1.3 and 4.x-HBase-1.2 branches?

**Pull request merged**